### PR TITLE
[FIX] mail: reopen emoji picker after emoji selection

### PR DESCRIPTION
**Pull request by alexkuhn**
Before this commit, when selecting an emoji from emoji picker, reopening emoji picker requires 2 click on the anchor.

This is due to `onSelect` that triggers the on close function of popover, but the internal state of `useEmojiPicker` was not aware. As a result, it thought emoji picker was still open, so 1st click was considered a close of emoji picker.

This commit fixes the issue by ensuring call to `onSelect` of emoji picker properly flags the emoji picker as closed.

### CODE REVIEW

File: addons/mail/static/tests/composer/composer_tests.js [hunk 1]
```diff
@@ -22,4 +24,25 @@ QUnit.module("mail", (hooks) => {
...
+                def.resolve();
+            },
+        });
+        const server = new TestServer();
+        server.addChannel(1, "general", "General announcements...");
+        const env = makeTestEnv((route, params) => server.rpc(route, params));
+        const { Component: PopoverContainer, props } = registry.category('main_components').get('PopoverContainer');
+        await mount(PopoverContainer, target, { env, props });
+        await mount(Composer, target, { env, props: { threadId: 1, placeholder: "owl" } });
+        await click($(target).find("i[aria-label='Emojis']").closest('button')[0]);
```
Comment (fdardenne) [inline addons/mail/static/tests/composer/composer_tests.js]: I think that JQuery should be avoided because we tend to remove this lib

**Comment by ged-odoo** [comment]
omg, that was the problem! i sometimes had it, but then could not reproduce it

File: addons/mail/static/src/composer/emoji_picker.js [hunk 2]
```diff
@@ -64,6 +66,11 @@ export class EmojiPicker extends Component {
         });
     }
 
+    afterLoadEmoji() {
```
Comment (fdardenne) [inline addons/mail/static/src/composer/emoji_picker.js]: This is interresting ! I'm quite curious as to whether or not it's a good practice to create an empty function to facilitate testing.
In my opinion, it is the tests that should adapt to the code and not the opposite.

An alternative I have in mind is to patch `setup()`, call the `super` in it and call a lifecycle hook that comes after the `onWillStart` to make the defer

**Review by fdardenne — COMMENTED** [review]
Hello ! 
I was just passing by and had few things that came to my mind.

A rather generic question: wouldn't it be nice to enable web linter/prettify for your codebase?

File: addons/mail/static/tests/composer/composer_tests.js (hunk 1 above)
Comment (ged-odoo) [reply addons/mail/static/tests/composer/composer_tests.js]: jquery is ok (but not encouraged) in tests

File: addons/mail/static/src/composer/emoji_picker.js (hunk 2 above)
Comment (ged-odoo) [reply addons/mail/static/src/composer/emoji_picker.js]: in this case, i would personally just load emoji data manually in the test itself

File: addons/mail/static/tests/composer/composer_tests.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/tests/composer/composer_tests.js]: change done (use `.querySelector` instead)

File: addons/mail/static/src/composer/emoji_picker.js (hunk 2 above)
Comment (alexkuhn) [reply addons/mail/static/src/composer/emoji_picker.js]: change done (manually `loadEmojiData()` in test)

**Comment by alexkuhn** [comment]
> A rather generic question: wouldn't it be nice to enable web linter/prettify for your codebase?

I'll prepare a PR to make use of prettier/linter in `mail/` from `web/tooling` config.

**Comment by alexkuhn** [comment]
Made a PR for prettier/linter: https://github.com/odoo-dev/odoo/pull/1384
I've added the same commit just before the `[FIX]` one in this PR, so that the code changes are properly formatted.

**Pull request merged**